src/components/common/Stepper.vue

```diff
@@ -91,11 +95,28 @@ export default class Stepper extends Vue {
 
       case RouteNames.REGISTRATION_DEFINE_BUSINESS: return this.getRegistration.defineBusinessValid
       case RouteNames.REGISTRATION_PEOPLE_ROLES: return this.isAddPeopleAndRolesValid
-      case RouteNames.REGISTRATION_REVIEW_CONFIRM: return this.isRegistrationValid
+      case RouteNames.REGISTRATION_REVIEW_CONFIRM: return this.isRegistrationValid()
     }
     return false
   }
 
+  /** Check for define and people components are valid first
+  * Check for review component when the FileAndPay button is clicked */
+  private isRegistrationValid (): boolean {
+    if (this.isTypeFirm) {
+      if (!this.isPartialRegistrationValid) return false
+      if (!this.getClickFileAndPay) return this.isPartialRegistrationValid
+      return this.isFinalRegistrationValid
+    }
+    return this.isFinalRegistrationValid
+  }
```

Comment:
This is too much of a special case:
- firms only
- hard-coded previous step validity